### ISSUE TITLE
N++ hangs temporarily on startup if it tries to load something from a sleeping HDD

### ISSUE THREAD

**Issue by UserNaem** [notepad-plus-plus/notepad-plus-plus#4059]
### Description of the Issue
At startup, N++ will try to re-open all files from the previous session. If any of the files in previous session reside on a HDD that went to sleep due to inactivity, N++ will hang itself and the operating system until the HDD wakes up and gives the file(s).
### Steps to Reproduce the Issue
1. Open a file from your secondary hard disk drive in N++, then close N++ to have the session be auto-saved
2. Wait until your hard disk drive sleeps
3. Run N++

### Expected Behavior
N++ starts quickly.

### Actual Behavior
N++ hangs (and also makes mouse movement all laggy) until it can read the file(s) from the sleeping hard drive.

**Comment by sasumner** [comment]
Isn't this the same thing?:
STEPS: Buy 50m hose at store. Connect hose to spigot. Turn on spigot.
ACTUAL BEHAVIOR: It takes many seconds before water comes out of the end of the hose.
EXPECTED BEHAVIOR: Water instantly available.
Sometimes expectations may not be realistic.

**Comment by UserNaem** [comment]
Other applications don't hang the entire operating system while trying to access files from sleeping HDDs. Even Explorer will let you use other applications and its other windows while it's waiting for a sleeping HDD.

**Comment by sasumner** [comment]
So what kind of time are we talking about here?

**Comment by UserNaem** [comment]
Depends on how much time it'll take for the HDD to wake up, mine hanged the system for a total of 8 seconds.

**Comment by Andrewskiz** [comment]
I have the same issue, sometimes it takes even longer with my external USB hard drive. It would be nice if n++ waited to access the files in the  tabs that are open until they are actually clicked.

**Comment by Faq** [comment]
ref #5389

**Comment by MetaChuh** [comment]
@UserNaem 

notepad++ does not cache any files at a secretly hidden shadow copy location, so it will need the respective drive to be online, in order to be able to load a file.

if you want notepad++ to load them quicker, please go to your windows power options, select advanced, and disable `turn off hard disk after [x] seconds` .

**side note for better understanding**:
it is just like baking cookies.
if your oven is cold, because it has not been turned on yet, it will take you a longer time, until your cookies are baked.
on the other hand, if your oven is switched on and heated up in advance, you will experience a faster baking cookies than ever before 😉

**Comment by Andrewskiz** [comment]
I understand why there is the delay but it would be nice if there was an option to not load all the files in the background tabs immediately for those who want to allow their drive to go to sleep.

**Comment by MetaChuh** [comment]
@Andrewskiz 

in your case, where notepad++ is already open, and stays open, you could go to `settings > preferences > misc` and disable `file status auto-detection` as seen at the screenshot below, if you don't need a background check of all opened files on every focus change, after you return to notepad++ (switching between apps).

![Imgur](https://i.imgur.com/lUMBFCW.png)

you can also speed up your notepad++ start for idle drives, if you go to `settings > preferences > recent files history` and activate `don't check at launch time`, as seen at the screenshot below, if it is not set already.

![Imgur](https://i.imgur.com/u068NDo.png)

**Comment by MetaChuh** [comment]
update for @Andrewskiz 

a recent [turbo commit](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/5389/commits/3034ad09ea490874c76bbfd8f58aa5f991757201) from @SinghRajenM adds the option to activate the file status auto-detection only for the current tab, as seen at the screenshot below. 👍

so your case, if you might want the choice to only check for external file modification updates on your currently active tab, but not for any background tabs, might be solved in future releases by this commit.

![img](https://user-images.githubusercontent.com/14791461/53896583-4d528a00-405a-11e9-8390-979c8dff05d4.png)

**Comment by MetaChuh** [comment]
commit has been implemented in 7.6.6

please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html).

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)
